docs/contributors/getting-started.md

```diff
@@ -69,45 +65,137 @@ repository:
+but most popular are:
+
+- debugging through logging (recommended option)
+- classic debugging with breakpoints
+
+### Debugging through logging
+
+This approach provides very quick iterations and short feedback loop.
+It depends on placing multiple `pprint.log()` calls which will log
+messages in `.metals.log` file. Logged output can be watched by `tail -f .metals/metals.log`.
+
+```
+MetalsLanguageServer.scala:1841 params: DebugSessionParams [
+  targets = SingletonList (
+    BuildTargetIdentifier [
+      uri = "file:/HappyMetalsUser/metals/#metals/Compile"
+    ]
+  )
+  dataKind = "scala-attach-remote"
+  data = {}
+]
+```
+
+See [workspace logs](#workspace-logs) for more information.
+
+This approach can be used in 2 variants:
+
+- together with [manual testing](#manual-tests) when it's hard to write test
+  for some changes.
+- with [unit tests](#unit-tests)
+
+### Classic debugging
+
+Classic debugging is possible by the [JVM debugging mechanism](#jvm-debugging).
+Publish Metals locally, open a new project and configure debug settings.
+Then you can attach IDE with opened Metals repository to the debugged instance:
+
+- VSCode - add attach configuration to yours [launch.json](../editors/vscode#via-a-launchjson-configuration)
+
+  ```
+  {
+    "version": "0.2.0",
+    "configurations": [
+      // Attach debugger when running via:
+      // `-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=localhost:5005`
+      {
+        "type": "scala",
+        "request": "attach",
+        "name": "Attach debugger to Metals server",
+        "buildTarget": "metals",
+        "hostName": "localhost",
+        "port": 5005
+      }
+    ]
+  }
+  ```
+
+  Then pick such a defined configuration and run debug.
+
+  ![Attach debugger](https://imgur.com/ySAo6Um.png)
+
+- IntelliJ - Select Attach to the process and pick proper process from the list
+
+  ![Attach to the process](https://imgur.com/lHSl57l.png)
+
 ## Unit tests
 
-To run the unit tests open an sbt shell and run `unit/test`
+To run the unit tests open an sbt shell and run `unit/test`. However, this command will
+run all of the unit tests declared in `unit` module.
 
 ```sh
 sbt
+
 # (recommended) run a specific test suite, great for edit/test/debug workflows.
 > unit/testOnly tests.DefinitionSuite
+
 # run a specific test case inside the suite.
-> unit/testOnly tests.DefinitionSuite -- *exact-test-name*
+> unit/testOnly tests.DefinitionSuite -- <exact-test-name>
+
 # run unit tests, moderately fast but still a bit too slow for edit/test/debug workflows.
 > unit/test
+
 # run slow integration tests, takes several minutes.
 > slow/test
-# run presentation compiler tests, these are the quickest tests to run.
-> cross/test
-# run presentation compiler tests for all Scala versions.
-> +cross/test
-# (not recommended) run all tests, slow. It's better to target individual projects.
-> test
```

Comment:
I think it might be a good idea to still leave a note in here that just running `test` isn't recommended as it will take a *long* time.